test/_setup.js

```diff
@@ -2,36 +2,52 @@ const Joi = require('joi');
 const Jo = require('../lib/index');
 const chai = require('chai');
 const Assertion = chai.Assertion;
+const util = require('util');
 
-global.expect = chai.expect;
-global.assert = chai.assert;
+const expect = global.expect = chai.expect;
+const assert = global.assert = chai.assert;
 
 
-function prettyPrintSchema (schema) {
-    return schema._rules.map((rule) => {
+function prettyPrintSchema(schema) {
+    return schema.getRules().map((rule) => {
         const args = rule._params
-                .map((param) => JSON.stringify(param))
-                .join(', ');
+        .map((param) => JSON.stringify(param))
+        .join(', ');
 
         return `${rule.name()}(${args})`;
     }).join('\n');
 }
 
-let validFns = [];
+function getNthCaller(n) {
+    const stack = new Error().stack.split('\n').slice(1);
+    const parts = (/[a-z\.]+\.js:[0-9]+/i).exec(stack[n])
+    return parts[0];
+}
+
+function pad(str, length, padder) {
+    while (str.length < length) {
+        str += padder || ' ';
+    }
 
-chai.use(function (_chai, utils) {
-    Assertion.addMethod('failOn', function (value, details) {
+    return str;
+}
+
+const validFns = [];
+
+chai.use((_chai, utils) => {
+    Assertion.addMethod('failOn', function (value, details, ignoreCompability, options) {
```

Comment:
The arguments after `value` should probably all be moved into an options object.